plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1963,8 +1963,10 @@ producer_plugin::get_unapplied_transactions_result producer_plugin::get_unapplie
 
 block_timestamp_type producer_plugin_impl::calculate_pending_block_time() const {
    const chain::controller& chain = chain_plug->chain();
-   const fc::time_point     now   = fc::time_point::now();
-   const fc::time_point     base  = std::max<fc::time_point>(now, chain.head().block_time());
+   // on speculative nodes, always use next block time. On producers, honor current clock time
+   const fc::time_point base  = _producers.empty()
```

Comment:
Not sure whether or not `chain.is_producer_node()` more clear than `_producers.empty()`.

Reply:
The `producer_plugin` uses `_producers.empty()` in a number of places for this concept. I could wrap that in a `producer_plugin` method if desired, but I don't think we should reach into controller for that.

Reply:
It is better to wrap that in a method in producer_plugin and modify other places. That makes the concept more clearer. But the change is not necessary for the PR.

Reply:
https://github.com/AntelopeIO/spring/pull/1034/commits/9f3c603e1ab24730beb764d63aca8b8ff57729eb